homeassistant/core.py

```diff
@@ -103,10 +104,20 @@
 # How long we wait for the result of a service call
 SERVICE_CALL_LIMIT = 10  # seconds
 
-# Source of core configuration
-SOURCE_DISCOVERED = "discovered"
-SOURCE_STORAGE = "storage"
-SOURCE_YAML = "yaml"
+
+class ConfigSource(StrEnum):
+    """Source of core configuration."""
+
+    DEFAULT = "default"
+    DISCOVERED = "discovered"
+    STORAGE = "storage"
+    YAML = "yaml"
+
+
+# SOURCE_* are deprecated as of Home Assistant > 2021.1, use ConfigSource instead
```

Comment:
Shouldn't this be 2022.1?

Reply:
2022.2 actually 😉 

```suggestion
# SOURCE_* are deprecated as of Home Assistant > 2022.2, use ConfigSource instead
```

Reply:
Actually I meant to write `> 2021.12` there, that would work (note `>`, not `>=`) even if for some reason there would be 2022.1.

Reply:
2022.2 is the first stable release this will be affected. That is how we document these things.